src/codegen/forward.jl

```diff
@@ -53,7 +53,8 @@ function fwd_transform!(ci, mi, nargs, N)
             return ZeroBundle{N}(mi.sparam_vals[stmt.args[1]::Int])
         elseif isexpr(stmt, :foreigncall)
             return Expr(:call, error, "Attempted to AD a foreigncall. Missing rule?")
-        elseif isexpr(stmt, :meta) || isexpr(stmt, :inbounds)  || isexpr(stmt, :loopinfo)
+        elseif isexpr(stmt, :meta) || isexpr(stmt, :inbounds)  || isexpr(stmt, :loopinfo) ||
+               isexpr(stmt, :boundscheck) || isexpr(stmt, :code_coverage_effect)
```

Comment:
`Expr(:boundscheck)` returns a bool, so this is incorrect.

Reply:
Fixed (and test added) in https://github.com/JuliaDiff/Diffractor.jl/pull/177